### PR TITLE
Add MinimumRequiredNumValidators

### PR DESCRIPTION
**Pull request by Kourin1996**
This PR adds MinimumRequiredNumValidators to prevent contract from removing required validators. Currently this value is set to hardcoded parameter MinimumRequiredNumValidators = 4.

This is forked from #2, please check the change at https://github.com/0xPolygon/staking-contracts/pull/3/commits/cb2e3eee5fc1ea90cbfc2daadd22be0e54c97305

### CODE REVIEW

**Review by lazartravica — APPROVED** [review]

File: contracts/staking.sol [hunk 1]
```diff
@@ -56,20 +68,25 @@ contract Staking {
         emit Staked(msg.sender, msg.value);
     }
 
-    function unstake() public onlyStaker {
+    function _unstake() private {
+        require(
+            _validators.length > MinimumRequiredNumValidators,
```
Comment (Vuksan) [inline contracts/staking.sol:73]: Maybe >=?
```suggestion
            _validators.length >= MinimumRequiredNumValidators,
```

File: test/staking.ts [hunk 2]
```diff
@@ -92,14 +93,35 @@ describe("Staking contract", function () {
...
-      await expect(contract.connect(accounts[1]).unstake()).to.be.revertedWith(
-        "Only staker can call function"
+      await expect(
+        contract.connect(accounts[numInitialValidators]).unstake()
+      ).to.be.revertedWith("Only staker can call function");
+    });
+
+    it("should failed if current number of validators equals to MinimumRequiredNumValidators", async () => {
+      // remove 1 account first
+      await contract.connect(accounts[1]).unstake();
```
Comment (Vuksan) [inline test/staking.ts:117]: I think there is no need to remove this account here.

**Review by Vuksan — CHANGES_REQUESTED** [review]

File: contracts/staking.sol (hunk 1 above)
Comment (Kourin1996) [reply contracts/staking.sol:73]: @Vuksan 

This condition is making sure that the number of validators doesn't go below `MinimumRequiredNumValidators`. If the number of current validators is `MinimumRequiredNumValidators`, any validator can't unstake.

File: test/staking.ts (hunk 2 above)
Comment (Kourin1996) [reply test/staking.ts:117]: I'm afraid I don't think so, the purpose of this test is making sure any validator can't unstake if the number of validator equals to `MinimumRequiredNumValidators`. At the beginning of the test, the number is 5 while MinimumRequiredNumValidators is 4. We need to decrease the number first.

File: contracts/staking.sol (hunk 1 above)
Comment (Vuksan) [reply contracts/staking.sol:73]: You're right 🙂

**Review by Vuksan — APPROVED** [review]
👏

**Pull request merged**